src/passes/OptimizeInstructions.cpp

```diff
@@ -767,6 +767,18 @@ struct OptimizeInstructions
     } else if (auto* iff = curr->dynCast<If>()) {
       iff->condition = optimizeBoolean(iff->condition);
       if (iff->ifFalse) {
+        if (auto* binary = iff->condition->dynCast<Binary>()) {
+          if (binary->op == NeInt64) {
+            if (auto* c = binary->right->dynCast<Const>()) {
+              if (c->value.geti64() == 0LL) {
+                // if((int64)x != 0) y else z  ==>  if(!x) z else y
```

Comment:
is this an i64 version of something we have an i32 optimization for? if so could this be alongside it, or even merged with it using Abstract etc.?

Reply:
No, it's special only for 64-bit. For 32-bits it's already handled in `optimizeBoolean` and [slightly differently](https://github.com/WebAssembly/binaryen/blob/765c295fda721296637b841201c057c69a579a4d/src/passes/OptimizeInstructions.cpp#L1021)

Reply:
Added comment about this